src/AddressSpace.h

```diff
@@ -242,6 +245,9 @@ class AddressSpace : public HasTaskSet {
     // equal to 'map'.
     const KernelMapping recorded_map;
     const EmuFile::shr_ptr emu_file;
+    // If this mapping has been mapped into the local address space, the address
+    // is recorded here, NULL otherwise.
+    void* local_addr;
```

Comment:
One question is whether the local mapping should always be read/write or whether it can be read-only or even PROT_NONE. Currently write_mem can write to memory regardless of tracee PROT_ status, so I think we should require that the local_addr be read-write if we can.